### PR TITLE
feature/kapersky_endpoint_security_add_new_fields

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/integration/issues/1792

### Added

- Add parsing support for `Result description: Processing error` events by extending `set_not_processed_fields` filter (shared stage)
- Newly populated ECS fields for `Processing error` events (existing field names, no schema addition):
	- `event.action`
	- `event.category`
	- `event.module`
	- `event.reason`
	- `event.type`
	- `file.directory`
	- `file.name`
	- `process.executable`
	- `process.pid`
	- `user.domain`
	- `user.name`

### Changed

- Modify parsing logic for existing ECS fields in `set_not_processed_fields`:
    - `event.reason`
    - `process.executable`
    - `user.domain`
    - `user.name`
- Improve parsing for `Not processed` events when optional fields are missing
- Improve user parsing robustness for identities with spaces in domain names (for example `NT AUTHORITY\\SYSTEM`)

### Fixed

- Fix `TypeError` in `set_not_processed_fields` when `Reason` is missing by making `event.reason` null-safe
- Fix `TypeError` in `set_not_processed_fields` when `Application path` or `Name` is missing by making `process.executable` null-safe
- Normalize `process.executable` path composition for `Not processed` events

## Type of Change

- [ ] New intake format
- [x] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product: Kaspersky/kaspersky_endpoint_security

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Extend Kaspersky Endpoint Security intake parsing for additional not-processed event types and harden field extraction for edge cases.

New Features:
- Support parsing of Kaspersky events with `Result description: Processing error` in the shared `set_not_processed_fields` stage.
- Populate additional ECS fields for not-processed and processing-error events, including event, file, process, and user attributes.

Bug Fixes:
- Prevent TypeErrors in `set_not_processed_fields` when `Reason`, `Application path`, or `Name` fields are missing in incoming events.
- Normalize and correct `process.executable` path composition for not-processed events to handle partial path data.

Enhancements:
- Improve parsing robustness for not-processed events when optional fields are absent, including safer construction of composite fields.
- Make user identity parsing more resilient to domains with spaces and special formats (for example `NT AUTHORITY\\SYSTEM`).

Documentation:
- Document the new behavior and field mappings for not-processed and processing-error events in the intake CHANGELOG.

Tests:
- Add parser test fixtures for not-processed events missing reason or process identity fields and for processing-error events.

### CODE REVIEW

File: Kaspersky/kaspersky_endpoint_security/tests/test_processing_error.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "input": {
+    "message": "Event type: Processing error\\r\\nName: WINWORD.EXE\\r\\nApplication path: C:\\Program Files\\Microsoft Office\\root\\Office16\\r\\nProcess ID: 12820\\r\\nUser: EXAMPLE\\user.test (Initiator)\\r\\nComponent: File Threat Protection\\r\\nResult description: Processing error\\r\\nObject type: File\\r\\nPath to object: \\\\server.example.local\\profiles$\\user.test\\Outlook\\r\\nObject name: user.test@example.org.ost\\r\\nReason: Read error"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Kaspersky/kaspersky_endpoint_security/tests/test_processing_error.json]: Fixed with commit [Apply changes requested by Copilot](https://github.com/SEKOIA-IO/intake-formats/pull/2757/changes/2d322663f393b34503dc0d53b7ecb8cda1cbe24e)

File: Kaspersky/kaspersky_endpoint_security/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+	- `event.category`
+	- `event.module`
+	- `event.reason`
+	- `event.type`
+	- `file.directory`
+	- `file.name`
+	- `process.executable`
+	- `process.pid`
+	- `user.domain`
+	- `user.name`
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Kaspersky/kaspersky_endpoint_security/CHANGELOG.md]: Fixed with commit [Apply changes requested by Copilot](https://github.com/SEKOIA-IO/intake-formats/pull/2757/changes/2d322663f393b34503dc0d53b7ecb8cda1cbe24e)

**Review by mchupeau-sk — DISMISSED** [review]
LGFM

**Review by mchupeau-sk — APPROVED** [review]
